public/app/features/scenes/variables/variants/CustomVariable.test.ts

```diff
@@ -0,0 +1,253 @@
+        },
+      ]);
+    });
+
+    it('Should generate correctly the options for only values as URLs', async () => {
+      const variable = new CustomVariable({
+        name: 'test',
+        options: [],
+        value: '',
+        text: '',
+        query: 'http://www.google.com/, http://www.amazon.com/',
+      });
+
+      await lastValueFrom(variable.validateAndUpdate());
+
+      expect(variable.state.value).toEqual('http://www.google.com/');
+      expect(variable.state.text).toEqual('http://www.google.com/');
+      expect(variable.state.options).toEqual([
+        {
+          label: 'http://www.google.com/',
+          value: 'http://www.google.com/',
+        },
+        {
+          label: 'http://www.amazon.com/',
+          value: 'http://www.amazon.com/',
+        },
+      ]);
+    });
+
+    it('Should generate correctly the options for key/values as URLs', async () => {
+      const variable = new CustomVariable({
+        name: 'test',
+        options: [],
+        value: '',
+        text: '',
+        query: 'google : http://www.google.com/, amazon : http://www.amazon.com/',
+      });
+
+      await lastValueFrom(variable.validateAndUpdate());
+
+      expect(variable.state.value).toEqual('http://www.google.com/');
+      expect(variable.state.text).toEqual('google');
+      expect(variable.state.options).toEqual([
+        {
+          label: 'google',
+          value: 'http://www.google.com/',
+        },
+        {
+          label: 'amazon',
+          value: 'http://www.amazon.com/',
+        },
+      ]);
+    });
+  });
+
+  describe('When value is provided', () => {
+    it('Should keep current value if current value is valid', async () => {
+      const variable = new CustomVariable({
+        name: 'test',
+        options: [],
+        query: 'A,B',
+        value: 'B',
+        text: 'B',
+      });
+
+      await lastValueFrom(variable.validateAndUpdate());
+
+      expect(variable.state.value).toBe('B');
+      expect(variable.state.text).toBe('B');
+    });
+
+    it('Should maintain the valid values when multiple selected', async () => {
+      const variable = new CustomVariable({
+        name: 'test',
+        options: [],
+        isMulti: true,
+        query: 'A,C',
+        value: ['A', 'B', 'C'],
+        text: ['A', 'B', 'C'],
+      });
+
+      await lastValueFrom(variable.validateAndUpdate());
+
+      expect(variable.state.value).toEqual(['A', 'C']);
+      expect(variable.state.text).toEqual(['A', 'C']);
+    });
+
+    it('Should pick first option if none of the current values are valid', async () => {
+      const variable = new CustomVariable({
+        name: 'test',
+        options: [],
+        isMulti: true,
+        query: 'A,C',
+        value: ['D', 'E'],
+        text: ['E', 'E'],
+      });
+
+      await lastValueFrom(variable.validateAndUpdate());
+
+      expect(variable.state.value).toEqual(['A']);
+      expect(variable.state.text).toEqual(['A']);
+    });
+  });
+});
```

Comment:
Technically these cases are already covered by tests on the base class (See tests on MultiValueVariable), but I guess it does not hurt :)

Reply:
Yep, I agree! I was doubting if adding them or not, but I decided to keep them there to make sure the CustomVariable keeps doing what is expected to do, no strong preference either :)